code/templates/Complex.hpp

```diff
@@ -1,6 +1,8 @@
 #include <ostream>
+#include <concepts>
 
 template <typename T=float>
+    requires std::floating_point<T>
```

Comment:
I intentionally did not write it as:
```c++
template <std::floating_point T=float>
```
So it's less unknown syntax for the students.

Reply:
Do we support concepts? This might not compile for some students.

A way out would be a feature test, but that might confuse them even more? 🤔

Reply:
The exercise already compiled with C++20 before, so I thought it was fine.

But this change breaks the solution, which also does `OrderedVector<Complex_t<Complex>>`, and `Complex` is not a floating point. Hence the question whether we actually want to support this below.

Reply:
I removed it again now. The `Comple_t<Complex>` is a bonus exercise, so the `requires` clause would ruin it.